code/data_structures/tree/binary_search_tree/binary_search_tree.cpp

```diff
@@ -1,88 +1,377 @@
-#include<iostream>
-using namespace std;
+/*
+** The C++ Implementation of Binary Search Tree
+**
+** Binary Search Tree, is a node-based binary tree data 
+** structure which has the following properties:
+**
+** - The left subtree of a node contains only nodes with 
+** keys less than the node’s key.
+** - The right subtree of a node contains only nodes with 
+** keys greater than the node’s key.
+** - The left and right subtree each must also be a binary 
+** search tree.
+*/
 
-struct node
+#include <iostream>
+
+using std::cout;
+using std::endl;
```

Comment:
Not recommended practice, so don’t use this.

Reply:
I don't agree with you here. It makes no sense to connect all of std namespace because of only two functions.

Reply:
I agree! I’m just saying to use std:: every time, as per recommended practice.